kmk/modules/dynamic_sequences.py

```diff
@@ -0,0 +1,253 @@
+from micropython import const
+from supervisor import ticks_ms
+
+from kmk.keys import KC, make_argumented_key
+from kmk.kmktime import check_deadline, ticks_diff
+from kmk.modules import Module
+
+
+class DMMeta:
+    def __init__(self, sequence_select=None):
+        self.sequence_select = sequence_select
+
+
+class SequenceStatus:
+    STOPPED = const(0)
+    RECORDING = const(1)
+    PLAYING = const(2)
+    SET_REPEPITIONS = const(3)
+    SET_INTERVAL = const(4)
+
+
+# Keycodes for number keys
+_numbers = range(KC.N1.code, KC.N0.code + 1)
+
+
+class Sequence:
+    def __init__(self):
+        self.repetitions = 1
+        self.interval = 0
+        self.sequence_data = [(set(), 0), (set(), 0), (set(), 0)]
```

Comment:
opaque data structure, suggestion: `collections.namedtuple` with `keys` and `timestamp` entries.